src/newViews/Tagging/Sources/LeadEditModal/index.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useCallback, useState } from 'react';
+import {
+    _cs,
+    isDefined,
+} from '@togglecorp/fujs';
+import {
+    Card,
+    Button,
+    Modal,
+    PendingMessage,
+} from '@the-deep/deep-ui';
+import {
+    useForm,
+} from '@togglecorp/toggle-form';
+
+import { useRequest, useLazyRequest } from '#utils/request';
+import { transformErrorToToggleFormError } from '#rest';
+
+import LeadEditForm from './LeadEditForm';
+import { schema, PartialFormType, Lead } from './LeadEditForm/schema';
+import styles from './styles.scss';
+
+// TODO: Show attachment's title and link if lead is attachment type
+
+interface Props {
+    className?: string;
+    onClose: () => void;
+    leadId?: number;
+    projectId: number;
+    onLeadSaveSuccess: () => void;
+}
+
+function LeadEditModal(props: Props) {
+    const {
+        className,
+        onClose,
+        projectId,
+        leadId,
+        onLeadSaveSuccess,
+    } = props;
+
+    const [initialValue, setInitialValue] = useState<PartialFormType>(() => ({
+        project: projectId,
+        sourceType: 'website',
+        priority: 100,
+        confidentiality: 'unprotected',
+    }));
+
+    const {
+        pristine,
+        value,
+        setFieldValue,
+        setValue,
+        error: riskyError,
+        validate,
+        setError,
+    } = useForm(schema, initialValue);
+
+    const {
+        pending: leadGetPending,
+    } = useRequest<Lead>({
+        skip: !leadId,
+        url: `server://v2/leads/${leadId}/`,
+        onSuccess: (response) => {
+            setInitialValue(response);
+            setValue(response);
+        },
+        failureHeader: 'Leads',
+    });
+
+    const {
+        pending: leadSavePending,
+        trigger: triggerLeadSave,
+    } = useLazyRequest<Lead, PartialFormType>({
+        url: leadId ? `server://v2/leads/${leadId}/` : 'server://v2/leads/',
+        method: leadId ? 'PATCH' : 'POST',
+        body: ctx => ({
+            ...ctx,
+            authors: ctx.authors ?? [],
+            emmTriggers: ctx.emmTriggers ?? [],
+            emmEntities: ctx.emmEntities ?? [],
```

Comment:
on need to do this. form can handle this

Reply:
how?